### PR TITLE
Fix detect AC3+ audio track, fixes#5684

### PR DESCRIPTION
**Pull request by SqTER-PL**

### CODE REVIEW

File: src/input/mpegts/dvb_psi_pmt.c [hunk 1]
```diff
@@ -359,6 +359,10 @@ dvb_psi_parse_pmt
     case 0x24:
       hts_stream_type = SCT_HEVC;
       break;
+          
+    case 0x87:
```
Comment (Trujulu) [inline src/input/mpegts/dvb_psi_pmt.c]: What about putting it after 0x81 to follow order?

**Comment by Trujulu** [comment]
P.S.: Does we need add 0x87 here too? Or is not needed? https://github.com/tvheadend/tvheadend/blob/771dfd6bea7bd4035ed991eccbe735dc00d3f800/src/descrambler/capmt.c#L2093

File: src/input/mpegts/dvb_psi_pmt.c (hunk 1 above)
Comment (SqTER-PL) [reply src/input/mpegts/dvb_psi_pmt.c]: You're right.
It can be after 0x81

Comment (SqTER-PL) [reply src/input/mpegts/dvb_psi_pmt.c]: I updated this file.
Now it's OK.

**Comment by SqTER-PL** [comment]
I corrected the file "capmt.c" according to your suggestion.
AC3+(EAC3) works without this change but the source code must be correct.
This will prevent an unexpected failure.

Look at the code now. I think it will be OK.

**Comment by Trujulu** [comment]
> I corrected the file "capmt.c" according to your suggestion.
> AC3+(EAC3) works without this change but the source code must be correct.
> This will prevent an unexpected failure.
> 
> Look at the code now. I think it will be OK.

I am not 100% sure of the capmt changes. They may be necessary, maybe not.

**Comment by SqTER-PL** [comment]
There was a mistake in the source code.
AC3 has estype = 0x81, and EAC3 estype = 0x87 (checked in the standard documentation).
This is certainly bad, but maybe not bother.
https://github.com/tvheadend/tvheadend/blob/771dfd6bea7bd4035ed991eccbe735dc00d3f800/src/descrambler/capmt.c#L2093

**Comment by SqTER-PL** [comment]
I will check the last change and let you know.

**Comment by SqTER-PL** [comment]
Check other estype. In all entries, the values for a given type are the same. It certainly was a mistake that needs to be corrected.

I have just checked and all changes work well.
Tvheadent now correctly detects EAC3 audio.
I searched the entire source code and found nothing to fix.

**Comment by Flole998** [comment]
Can you squash this into one commit please? Other than that: looks good to me

**Pull request closed without merging**